chrome/content_script.js

```diff
@@ -891,14 +901,19 @@ chrome.runtime.onMessage.addListener(
       passwordalert.start_(msg);
     }
 );
-document.addEventListener('DOMContentLoaded', function() {
-  passwordalert.domContentLoaded_ = true;
-  passwordalert.completePageInitializationIfReady_();
-});
-// Check to see if we already missed DOMContentLoaded:
-if (document.readyState == 'interactive' ||
-    document.readyState == 'complete' ||
-    document.readyState == 'loaded') {
-  passwordalert.domContentLoaded_ = true;
-  passwordalert.completePageInitializationIfReady_();
+
+domReadyCheck = function() {
+  var loginForm = document.querySelector('#view_container > form');
+  if (loginForm && document.getElementById('Email')) {
+    passwordalert.domContentLoaded_ = true;
+  } else if (document.getElementById('hiddenEmail') &&
+             document.getElementsByName('password')){
+    passwordalert.domContentLoaded_ = true;
+  }
+  if (passwordalert.domContentLoaded_) {
+    window.removeEventListener('DOMNodeInserted', domReadyCheck);
+    passwordalert.completePageInitializationIfReady_();
```

Comment:
completePageInitializationIfReady_() handles other types of login pages (sso and forced change) in addition to the main login page. So gating completePageInitializationIfReady_() behind a check for document.querySelector('#view_container > form') will probably not work for those cases.

Reply:
You're right, good catch. We don't use sso in our managed policy so we missed that part. What do you think would make the most sense here to fix this issue?